src/values/ECMAScriptSourceFunctionValue.js

```diff
@@ -23,6 +23,7 @@ export default class ECMAScriptSourceFunctionValue extends ECMAScriptFunctionVal
   $Strict: boolean;
   $FormalParameters: Array<BabelNodeLVal>;
   $ECMAScriptCode: BabelNodeBlockStatement;
+  $IsSimpleReactClassComponent: ?boolean;
```

Comment:
This isn't really that nice for pluggability. E.g. other plugins won't be able to have the same flexibility to keep adding flags.

What do you think about using a `WeakSet` in `react/reconciliation` instead?

Any `ObjectValue` won't be particularly fast anyway since they have so many properties. This will be a map-like expando/lookup regardless. So a small `WeakSet` might actually also be as fast or faster.

Reply:
I switched to a `WeakMap` because we need to know if the type is `"SIMPLE" | "COMPLEX"`. Having a set wouldn't give us this data as there are three routes:
- not yet evaluated to be either simple or complex
- being simple
- being complex